net/tailscale/files/tailscale.init

```diff
@@ -1,47 +1,250 @@
 
-  config_load tailscale
-  config_get_bool std_out "settings" log_stdout 1
-  config_get_bool std_err "settings" log_stderr 1
-  config_get port "settings" port 41641
-  config_get state_file "settings" state_file /etc/tailscale/tailscaled.state
-  config_get fw_mode "settings" fw_mode nftables
+service_triggers() {
+  procd_add_reload_trigger "tailscale"
+  procd_add_interface_trigger "interface.*.up" wan /etc/init.d/tailscale reload
+}
 
-  /usr/sbin/tailscaled --cleanup
+section_enabled() {
+  config_get_bool enabled "$1" 'enabled' 0
+  [ $enabled -gt 0 ]
+}
+
+custom_instance() {
+  local cfg="$1"
+  local acceptRoutes hostname acceptDNS advertiseExitNode exitNode advertiseRoutes s2s subnetRoutes flags loginServer authkey std_out std_err
+  local ARGS=" up --reset"
+
+  config_get_bool acceptRoutes $cfg 'acceptRoutes'
+  config_get hostname $cfg 'hostname'
+  config_get_bool acceptDNS $cfg 'acceptDNS'
+  config_get_bool advertiseExitNode $cfg 'advertiseExitNode'
+  config_get exitNode $cfg 'exitNode'
+  config_get advertiseRoutes $cfg 'advertiseRoutes'
+  config_get_bool s2s $cfg 's2s'
+  config_get flags $cfg 'flags'
+  config_get loginServer $cfg 'loginServer'
+  config_get authkey $cfg 'authkey'
+  config_get_bool std_out $cfg 'log_stdout'
+  config_get_bool std_err $cfg 'log_stderr'
+
+  [ "$acceptRoutes" = "1" ] && ARGS="$ARGS --accept-routes=true"
+  [ -n "$hostname" ] && ARGS="$ARGS --hostname=$hostname"
+  [ "$acceptDNS" = "0" ] && ARGS="$ARGS --accept-dns=false"
+  [ "$advertiseExitNode" = "1" ] && ARGS="$ARGS --advertise-exit-node"
+  [ -n "$exitNode" ] && ARGS="$ARGS --exit-node=$exitNode --exit-node-allow-lan-access=true"
+  [ -n "$advertiseRoutes" ] && ARGS="$ARGS --advertise-routes=$(echo $advertiseRoutes | tr ' ' ',')"
+  [ "$s2s" = "1" ] && ARGS="$ARGS --snat-subnet-routes=false"
+  [ -n "$flags" ] && ARGS="$ARGS $flags"
+  [ -n "$loginServer" ] && ARGS="$ARGS --login-server=$loginServer"
+  [ -n "$authkey" ] && ARGS="$ARGS --authkey=$authkey"
 
   procd_open_instance
-  procd_set_param command /usr/sbin/tailscaled
+  procd_set_param command $PROG $ARGS
+  procd_set_param stdout "$std_out"
+  procd_set_param stderr "$std_err"
+  procd_close_instance
+  (
+    [ -f "/var/run/tailscale.wait.pid" ] && return
+    touch /var/run/tailscale.wait.pid
+    count=0
+    while [ -z "$(ifconfig | grep 'tailscale' | awk '{print $1}')" ] || [ -z "$(tailscale ip -4)" ]
+    do
+      sleep 2
+      let count++
+      [ "${count}" -ge 5 ] && { rm /var/run/tailscale.wait.pid; exit 19; }
+    done
 
-  # Starting with v1.48.1 ENV variable is required to enable use of iptables / nftables.
-  # Use nftables by default - can be changed to 'iptables' in tailscale config
-  procd_set_param env TS_DEBUG_FIREWALL_MODE="$fw_mode"
+    if [ "$acceptDNS" = "1" ]; then
+      MagicDNSSuffix=$(tailscale status --json | awk -F'"' '/"MagicDNSSuffix"/ {last=$(NF-1)} END {print last}')
+      sed -i '/100.100.100.100/d' /etc/dnsmasq.conf
```

Comment:
`100.100.100.100` seems to be coming out of nowhere. Maybe put in some constant with comments?

Reply:
100.100.100.100 is the DNS server address that is hardcoded into tailscale as far as I know.